app/services/concerns/xccdf/benchmarks.rb

```diff
@@ -7,12 +7,15 @@ module Benchmarks
 
     included do
       def save_benchmark
-        ::Xccdf::Benchmark.import!([benchmark].select(&:new_record?),
-                                   ignore: true)
+        benchmark.package_name = package_name
+
+        return unless benchmark.new_record? || benchmark.package_name_changed?
+
+        benchmark.save!
```

Comment:
@marleystipich2 I think the `import!` was an overkill for conditionally inserting one record and it was not really respecting the update of the package name. So I am changing it to a conditional save if the record is new OR the package name is being changed.

Reply:
@skateman are we sure that the `benchmark.new_record?` condition would cover previous `ignore: true` to prevent unique constraint failure?

Reply:
I don't see why it would behave differently from its [documentation](https://apidock.com/rails/ActiveRecord/Base/new_record%3F) and the `from_openscap_parser` method uses `find_or_initialize_by`

Reply:
If `find_or_initialize_by` is used then all is fine, unless it is created concurrently in between. `#new_record?` does not check object existence within db or does it? A new conflicting model instance can be initialized (having `new_record?` true) and then tried to be saved.

Reply:
Well, I don't see how the import would help us with the package name :confused: maybe I could remove the `!` from the `save` that would ensure integrity, or catch the duplicate exception and ensure that the package name gets updated anyway.

Reply:
Maybe an upsert? Anyhow, I guess it is good as it is. The concurrency might be ignored, for now.

Reply:
Right, we have rails6 :see_no_evil:

Reply:
Okay, keeping it as it is.